src/portfolios/dto/set-custodian.dto.ts

```diff
@@ -0,0 +1,23 @@
+/* istanbul ignore file */
+
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { IsDate } from 'class-validator';
+
+import { IsDid } from '~/common/decorators/validation';
+import { TransactionBaseDto } from '~/common/dto/transaction-base-dto';
+
+export class SetCustodianDto extends TransactionBaseDto {
+  @ApiProperty({
+    description: 'Th DID of identity to be set as custodian',
+    example: '0x0600000000000000000000000000000000000000000000000000000000000000',
+  })
+  @IsDid()
+  readonly target: string;
+
+  @ApiPropertyOptional({
+    description: 'Expiry date for the custody over Portfolio',
+    example: new Date('05/23/2021').toISOString(),
+  })
+  @IsDate()
```

Comment:
```suggestion
  @IsOptional()
  @IsDate()
```

Pretty sure you need `IsOptional`, but I could be wrong.

There are two types of decorators being applied here, Swagger ones to generate the docs / interface, and then class-validator ones that will return BadRequest to the user.

class-validator can't infer `expiry` is optional since it knows neither Swagger nor our TS type.